### ISSUE TITLE
After upgrading from ABP 6.0.5 to 7.0.1, the custom mapping section of AutoMapper does not work.

### ISSUE THREAD

**Issue by hexianggui** [abpframework/abp#15850]
AutoMapperProfile:
`
CreateMap<DeviceUpdateDto, Device>(MemberList.Source)
            .ForMember(d => d.Images, o => o.MapFrom(s => string.Join(",", s.Images)));
`
Error Message:

```
Autofac.Core.DependencyResolutionException: An exception was thrown while activating Volo.Abp.BackgroundJobs.BackgroundJobStore -> Volo.Abp.ObjectMapping.DefaultObjectMapper`1[[Volo.Abp.BackgroundJobs.AbpBackgroundJobsDomainModule, Volo.Abp.BackgroundJobs.Domain, Version=7.0.2.0, Culture=neutral, PublicKeyToken=null]] -> Volo.Abp.AutoMapper.AutoMapperAutoObjectMappingProvider`1[[Volo.Abp.BackgroundJobs.AbpBackgroundJobsDomainModule, Volo.Abp.BackgroundJobs.Domain, Version=7.0.2.0, Culture=neutral, PublicKeyToken=null]] -> λ:Volo.Abp.AutoMapper.IMapperAccessor -> λ:Volo.Abp.AutoMapper.MapperAccessor.
 ---> AutoMapper.AutoMapperConfigurationException:
Unmapped members were found. Review the types and members below.
Add a custom mapping expression, ignore, add a custom resolver, or modify the source/destination type
For no matching constructor, add a no-arg ctor, add optional arguments, or map all of the constructor parameters
================================================================
DeviceCreateDto -> Device (Source member list)
Mobest.Mes.Devices.DeviceCreateDto -> Mobest.Mes.Devices.Device (Source member list)

Unmapped properties:
Images
================================================================
DeviceUpdateDto -> Device (Source member list)
Mobest.Mes.Devices.DeviceUpdateDto -> Mobest.Mes.Devices.Device (Source member list)

Unmapped properties:
Images

   at Volo.Abp.AutoMapper.AbpAutoMapperModule.<>c__DisplayClass2_1.<CreateMappings>g__ValidateAll|2(IConfigurationProvider config)
   at Volo.Abp.AutoMapper.AbpAutoMapperModule.CreateMappings(IServiceProvider serviceProvider)
   at Autofac.Extensions.DependencyInjection.AutofacRegistration.<>c__DisplayClass3_0.<Register>b__0(IComponentContext context, IEnumerable`1 parameters)
   at Autofac.Core.Activators.Delegate.DelegateActivator.ActivateInstance(IComponentContext context, IEnumerable`1 parameters)
   at Autofac.Core.Activators.Delegate.DelegateActivator.<ConfigurePipeline>b__2_0(ResolveRequestContext ctxt, Action`1 next)
   at Autofac.Core.Resolving.Middleware.DisposalTrackingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
   at Autofac.Core.Resolving.Middleware.ActivatorErrorHandlingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
   --- End of inner exception stack trace ---
   at Autofac.Core.Resolving.Middleware.ActivatorErrorHandlingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
   at Autofac.Core.Resolving.Middleware.SharingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
   at Autofac.Core.Resolving.Middleware.CircularDependencyDetectorMiddleware.Execute(ResolveRequestContext context, Action`1 next)
   at Autofac.Core.Resolving.ResolveOperation.GetOrCreateInstance(ISharingLifetimeScope currentOperationScope, ResolveRequest request)
   at Autofac.Core.Resolving.ResolveOperation.ExecuteOperation(ResolveRequest request)
   at Autofac.ResolutionExtensions.TryResolveService(IComponentContext context, Service service, IEnumerable`1 parameters, Object& instance)
   at Autofac.ResolutionExtensions.ResolveService(IComponentContext context, Service service, IEnumerable`1 parameters)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Volo.Abp.BackgroundJobs.BackgroundJobWorker.DoWorkAsync(PeriodicBackgroundWorkerContext workerContext)
   at Volo.Abp.BackgroundJobs.BackgroundJobWorker.DoWorkAsync(PeriodicBackgroundWorkerContext workerContext)
   at Volo.Abp.BackgroundWorkers.Hangfire.HangfirePeriodicBackgroundWorkerAdapter`1.DoWorkAsync(CancellationToken cancellationToken)
   at InvokeStub_TaskAwaiter.GetResult(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

**Comment by hexianggui** [comment]
The problem has been solved!
Add the following code to the host module

```
private void ConfigureAutoMapper()
        {
            Configure<AbpAutoMapperOptions>(options =>
            {
                options.AddMaps<MobestHttpApiHostModule>();
            });
        }
```
```
public override void ConfigureServices(ServiceConfigurationContext context){
......
ConfigureAutoMapper();
........
}
```

**Comment by berkansasmaz** [comment]
Interesting, could it be that the profile wasn't registered somehow? Can you check against [this doc](https://docs.abp.io/en/abp/latest/Object-To-Object-Mapping#define-mappings)? By the way, do the other mapping definitions you define work properly or is it just a problem when mapping this `DeviceUpdateDto` to `Device`?

**Comment by hexianggui** [comment]
> Interesting, could it be that the profile wasn't registered somehow? Can you check against [this doc](https://docs.abp.io/en/abp/latest/Object-To-Object-Mapping#define-mappings)? By the way, do the other mapping definitions you define work properly or is it just a problem when mapping this `DeviceUpdateDto` to `Device`?

6.0 is normal. This problem occurs only after upgrading to 7.0. I removed this image attribute and other DTO custom configurations will not work. I will verify it tomorrow

**Comment by berkansasmaz** [comment]
Thanks for the reply. After you verify tomorrow, if there seems to be a problem with the framework, I will perform my tests on `v7.0.0` and take action accordingly. I will be waiting for your reply. Have a nice day.

**Comment by hexianggui** [comment]
I've tested it, and none of the custom configurations for AutoMapper are working! I created a brand new project using the Module template, and it's the same issue. Projects created using the app template have no problems.
This is the project I created using the Module template: [tabpmapperTest](https://github.com/hexianggui/tabpmapperTest). I encountered the same error when running MapperTestAppService through Swagger.
You can either create a new project using the Module template or clone the [tabpmapperTest](https://github.com/hexianggui/tabpmapperTest) repository to reproduce these issues. Thank you.
```
[13:48:56 ERR] An exception was thrown while activating Volo.Abp.ObjectMapping.DefaultObjectMapper`1[[Motest.TEST.TESTApplicationModule, Motest.TEST.Application, Version=0.1.0.0, Culture=neutral, PublicKeyToken=null]] -> Volo.Abp.AutoMapper.AutoMapperAutoObjectMappingProvider`1[[Motest.TEST.TESTApplicationModule, Motest.TEST.Application, Version=0.1.0.0, Culture=neutral, PublicKeyToken=null]] -> λ:Volo.Abp.AutoMapper.IMapperAccessor -> λ:Volo.Abp.AutoMapper.MapperAccessor.
Autofac.Core.DependencyResolutionException: An exception was thrown while activating Volo.Abp.ObjectMapping.DefaultObjectMapper`1[[Motest.TEST.TESTApplicationModule, Motest.TEST.Application, Version=0.1.0.0, Culture=neutral, PublicKeyToken=null]] -> Volo.Abp.AutoMapper.AutoMapperAutoObjectMappingProvider`1[[Motest.TEST.TESTApplicationModule, Motest.TEST.Application, Version=0.1.0.0, Culture=neutral, PublicKeyToken=null]] -> λ:Volo.Abp.AutoMapper.IMapperAccessor -> λ:Volo.Abp.AutoMapper.MapperAccessor.
 ---> AutoMapper.AutoMapperConfigurationException:
Unmapped members were found. Review the types and members below.
Add a custom mapping expression, ignore, add a custom resolver, or modify the source/destination type
For no matching constructor, add a no-arg ctor, add optional arguments, or map all of the constructor parameters
======================================================
MapperCreateDto -> MapperTest (Source member list)
Motest.TEST.MapperCreateDto -> Motest.TEST.MapperTest (Source member list)

Unmapped properties:
Images

   at Volo.Abp.AutoMapper.AbpAutoMapperModule.<>c__DisplayClass2_1.<CreateMappings>g__ValidateAll|2(IConfigurationProvider config)
   at Volo.Abp.AutoMapper.AbpAutoMapperModule.CreateMappings(IServiceProvider serviceProvider)
   at Autofac.Core.Activators.Delegate.DelegateActivator.ActivateInstance(IComponentContext context, IEnumerable`1 parameters)
   at Autofac.Core.Activators.Delegate.DelegateActivator.<ConfigurePipeline>b__2_0(ResolveRequestContext ctxt, Action`1 next)
   at Autofac.Core.Resolving.Middleware.DisposalTrackingMiddleware.Execute(ResolveRequestContext context, Action`1 next)   at Autofac.Core.Resolving.Middleware.ActivatorErrorHandlingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
........
```

**Comment by berkansasmaz** [comment]
You can solve the problem by changing the `MapperCreateDto` to `MapperTest` configuration in your Profile file as below:

    CreateMap<MapperCreateDto, MapperTest>()
            .Ignore(x => x.Id)
           .ForMember(d => d.Images, o => o.MapFrom(s => string.Join(",", s.Images)));

**Comment by hexianggui** [comment]
Thank you very much, this configuration solves the problem. However, since my project already has a lot of code, it would be a bit cumbersome to reconfigure everything.

**Comment by berkansasmaz** [comment]
Sorry to hear that. I hope it won't cause too much trouble. By the way, it's best practice not to use auto object mapping for Input DTO to Entity mappings, FYI. 

Ref: https://docs.abp.io/en/abp/4.1/Domain-Driven-Design-Implementation-Guide#object-to-object-mapping

---

Closing the issue. Feel free to re-open or create a new issue if you have further questions.

**Comment by iamsedatozturk** [comment]
Thanks for the heads-up @berkansasmaz 
Can you check here? I think it uses input dto to entity mapping which you suggested to avoid...
https://docs.abp.io/en/abp/latest/Tutorials/Part-1?UI=NG&DB=EF
`CreateMap<CreateUpdateBookDto, Book>();`

**Comment by EngincanV** [comment]
> Thanks for the heads-up @berkansasmaz Can you check here? I think it uses input dto to entity mapping which you suggested to avoid... https://docs.abp.io/en/abp/latest/Tutorials/Part-1?UI=NG&DB=EF `CreateMap<CreateUpdateBookDto, Book>();`

Hi, we have mapped the input DTO to an entity, because it's a tutorial and it aims to develop an application fast as possible, therefore architectural concerns can be overlooked.  For your own applications, you should not map input DTOs to entities: [ttps://docs.abp.io/en/abp/4.1/Domain-Driven-Design-Implementation-Guide#object-to-object-mapping](https://docs.abp.io/en/abp/4.1/Domain-Driven-Design-Implementation-Guide#object-to-object-mapping)
